release-please-config.json

```diff
@@ -0,0 +1,68 @@
+{
+  "bootstrap-sha": "79f816980073ecabc07b81eeb307ee9e2024621b",
+  "packages": {
+    ".": {
+      "release-type": "ruby",
+      "version-file": "lib/metricks/version.rb",
+      "changelog-path": "CHANGELOG.md",
+      "bump-minor-pre-major": true,
+      "bump-patch-for-minor-pre-major": true,
+      "draft": false,
+      "prerelease": false
+    }
+  },
+  "changelog-sections": [
+    {
+      "type": "feat",
+      "section": "Features"
+    },
+    {
+      "type": "feature",
+      "section": "Features"
+    },
+    {
+      "type": "fix",
+      "section": "Bug Fixes"
+    },
+    {
+      "type": "perf",
+      "section": "Performance Improvements"
+    },
+    {
+      "type": "revert",
+      "section": "Reverts"
+    },
+    {
+      "type": "docs",
+      "section": "Documentation"
+    },
+    {
+      "type": "style",
+      "section": "Styles"
+    },
+    {
+      "type": "chore",
+      "section": "Miscellaneous Chores"
+    },
+    {
+      "type": "refactor",
+      "section": "Code Refactoring"
+    },
+    {
+      "type": "test",
+      "section": "Tests",
+      "hidden": true
+    },
+    {
+      "type": "build",
+      "section": "Build System",
+      "hidden": true
+    },
+    {
+      "type": "ci",
+      "section": "Continuous Integration",
+      "hidden": true
+    }
+  ],
```

Comment:
Since this is a public repo, I think we can remove this and fallback to the [default value](https://git.io/JqCZL), which adds `feat`, `fix`, `perf`, and `revert` types to the changelog.

We are unlikely to want to cut a new version of the gem for `docs`, `style`, `chore`, or `refactor` changes, as they should have no externally facing effect.

Reply:
We're having a similar conversation over here 😅  https://github.com/krystal/abe/pull/15#issuecomment-1558939598

Reply:
AFAIK if you don't set lib upgrades as one of the commit types considered for release, then release-please won't generate a new release version.

Reply:
@jimeh so are we using `deps` as the commit type now for upgrades of libs to trigger new release versions?

Reply:
I've removed that set of manifest changelog sections now

Reply:
`deps` is not a standard/default type, so we would need to include the full changelog sections list and add `deps` to it if we wanted to use that. Which I think is both a little annoying and probably overkill.

Upgrading dependencies are often a `chore`, and in theory, that does not merit creating a new release cause end-users get no and see change/benefit. Specially not for a gem for example if the upgrade is just increasing the minimal required version for "general maintenance" purposes. If anything, that just causes headaches for people who may be stuck with any of the no longer supported older versions of dependencies.

If you're upgrading a dependency list to exclude a version with a known security flaw, I would consider that a `fix` type, cause you're fixing a potential security vulnerability.

Likewise, updating dependencies to add compatibility with Rails 7 for example, I would see that as either a `feat` or a `fix` depending on the if incompatibility with Rails 7 was originally intentional or accidental.

Reply:
Thanks @jimeh so for clarity, the list of declared types in this list does actually influence what release-please considers to be a "releasable unit" is that correct?

Reply:
If so... I'm guess i'm wondering how we know if it gets released as a minor or patch version? I understand major will be incremented if we use `feat!:` or `fix!:` etc.

Reply:
@paulsturgess Yes, the `hidden` property within each section entry determines if the type shows in the changelog or not. And anything that shows in the changelog is considered a releasable unit.

Commits marked as containing breaking changes (via `!` or `BREAKING CHANGE:` footer) always show in the changelog regardless of commit type, so they are always a releasable unit.

The one unknown with that for me which I'm gonna test, is if you make up a commit type like `foobar!: oopsy`. I don't know if that shows, as it's not a supported commit type at all, even though it is parsable as a conventional commit.

Reply:
ok, so breaking change marked commits with completely made up types, do still show in the changelog:

![Screen-Shot-2023-05-23-12-32-11 21](https://github.com/krystal/metricks/assets/39563/99e2ce77-4d29-404b-8980-a9c9c3a9453b)

The commits here are:

```
chore!: break nothing
```

```
foobarnope!: this also does not break stuff
```

```
helloworld: also does not break stuff

BREAKING CHANGE: breaking things? nah
```